src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs

```diff
@@ -131,24 +132,33 @@ private static Metadata GetMetadataNoThrow(PortableExecutableReference reference
             CancellationToken cancellationToken)
         {
             var filePath = reference.FilePath;
+            var checksum = IOUtilities.PerformIO(() =>
+            {
+                using (var stream = File.OpenRead(filePath))
```

Comment:
you can use this (http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Execution/Serializer.cs,51)
or use the service directly (http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Execution/IReferenceSerializationService.cs,15)
to get checksum for metadata. it uses MVID so, it should be cheaper than creating checksum from whole content of the metadata